File: 118. Pascal-s Triangle/solution.go
func generate(numRows int) [][]int {
    if numRows == 0 {
        var sol = [][]int{}
        return sol
    } else if numRows == 1 {
        var sol = [][]int{{1}}
        return sol
    } else if numRows == 2 {
            var sol = [][]int{{1},{1,1}}
        return sol
    }
    var sol = [][]int{{1},{1,1}}

    for i:=2; i<numRows; i++ {
        var sol2 = []int{1}
        for j:=0; j<len(sol[i-1])-1; j++ {
            sol2 = append(sol2, sol[i-1][j] + sol[i-1][j+1])
        }
        sol2 = append(sol2, 1)
        sol = append(sol, sol2)
    }
    return sol
}

File: 118. Pascal-s Triangle/solution.py
class Solution:
    def generate(self, numRows: 'int') -> 'List[List[int]]':
        sol = []
        if numRows == 0:
            return sol
        sol.append([1])
        if numRows == 1:
            return sol
        sol.append([1,1])
        if numRows == 2:
            return sol
        
        for i in range(3, numRows+1):
            s = [sol[-1][0]]
            for j in range(1, len(sol[-1])):
            s.append(sol[-1][-1])
            sol.append(s)
            
        return sol
            
